src/ripple/app/misc/NegativeUNLVote.cpp

```diff
@@ -135,29 +112,28 @@ NegativeUNLVote::addTx(
     }
 }
 
-NodeID
+PublicKey
 NegativeUNLVote::choose(
     uint256 const& randomPadData,
-    std::vector<NodeID> const& candidates)
+    std::vector<PublicKey> const& candidates)
 {
     assert(!candidates.empty());
-    static_assert(NodeID::bytes <= uint256::bytes);
-    NodeID randomPad = NodeID::fromVoid(randomPadData.data());
-    NodeID txNodeID = candidates[0];
+    PublicKey chosenValPubKey = candidates[0];
     for (int j = 1; j < candidates.size(); ++j)
     {
-        if ((candidates[j] ^ randomPad) < (txNodeID ^ randomPad))
+        if ((*candidates[j].data() ^ *randomPadData.data()) <
```

Comment:
This is very wrong. You XOR the pointers so this will be different only every mahcine. Also if you change the NodeID to PublicKey will require amendment because it changes the selection logic for the candidate.

Reply:
1. I'm dereferencing the pointers, so I'm using the data contained within these pointers. I'm not using the actual pointer themselves.
2. Yes, this code change will require an amendment.

Reply:
I believe it requires an amendment because it changes the rules for Negative UNL selection. At the moment, Node IDs are used in the selection, but the specification and the documentation for NegativeUNL indicates that PublicKey is used for the same.